src/Microsoft.Azure.WebJobs.Extensions.Kafka/Output/KafkaAttributeBindingProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Confluent.Kafka;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka
+{
+    internal partial class KafkaAttributeBindingProvider : IBindingProvider
+    {
+        private static readonly IKafkaProducerBindingProvider InnerProvider = new CompositeKafkaProducerBindingProvider(
+            new AsyncCollectorArgumentBindingProvider(),
+            new KafkaEventDataArgumentBindingProvider(),
+            new StringArgumentBindingProvider(),
+            new ByteArrayArgumentBindingProvider(),
+            new SerializableTypeArgumentBindingProvider()
+            );
```

Comment:
This appears to be using our "old" method of writing these bindings, which some of our earlier bindings (SB/Storage/etc) all still use. There is a newer way to do this that doesn't require nearly as much code, and there is support for open generics. You can see an example in the cosmos binding for a collector, which starts here: https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.CosmosDB/Config/CosmosDBExtensionConfigProvider.cs#L56

There's also some documentation on how to do this here: https://github.com/Azure/azure-webjobs-sdk/wiki/Creating-custom-input-and-output-bindings#binding-to-generic-types-with-opentypes

That being said, this approach will work -- @fabiocav, does it matter if this extension is using the older method for wiring this stuff up?

Reply:
@brettsam  can we accept as is, and go back and refactor to the new way after? Or would this result in a breaking change? 

@fbeltrao how much effort to refactor to the new way now?

Reply:
By glancing at the new way of doing it I think that this is a small/medium change. Problem I have is the timing. 

If it is a must have I will need to fit in my schedule. However, if it is "nice to have" I suggest to accept this PR and plan the change for the "next release after Build". I believe that the required changes won't surface to the external API, so customers won't be affected by breaking changes.

Reply:
It shouldn't result in a breaking change -- most of the "new" way was built to reduce boilerplate code that every extension was writing. I'm not sure if it's a "must-have" to get this committed. As long as you have good tests in place for behavior, it reduces the chance of any inadvertent breaking change.

@fabiocav -- are you okay with them coming back later to use the new binding model?

Reply:
I'm onboard with the accept now and come back and update to new version, if everyone else is. Thoughts?